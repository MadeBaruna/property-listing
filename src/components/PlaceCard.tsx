import React from 'react';
import styled from 'styled-components';
import MdiCityIcon from 'mdi-react/CityIcon';

// this component inspired from: https://codepen.io/virgilpana/pen/RNYQwB

const Card = styled.div`
  display: flex;
  background: white;
  flex-direction: column;
  width: 335px;
  margin: 10px;
  cursor: pointer;
  transition: 100ms ease-out; 

  &:hover {
    box-shadow:0px 13px 21px -5px rgba(0, 0, 0, 0.2);
    transition:  100ms ease-out; 
  }
`;

const ImageContainer = styled.div`
  position: relative;

  img {
    object-fit: cover;
    width: 100%;
  }

  span {
    position: absolute;
    bottom: 10px;
    right: 10px;
    background: rgba(0, 0, 0, 0.5);
    padding: 5px 10px;
    color: white;
    opacity: 0;
    transition: 100ms ease-out; 

    ${Card}:hover & {
      opacity: 1;
      transition: 100ms ease-out; 
    }
  }
`;

const Content = styled.div`
  padding: 10px;

  h3 {
    white-space: nowrap;
    overflow: hidden;
    display: block;
    text-overflow: ellipsis;
  }

  p {
    height: 100%;
    line-height: 18px;
    max-height: 36px;
    word-wrap: break-word;
    overflow: hidden;
    margin-top: 15px;
    display: flex;

    &.city {
      align-items: center;
    }
  }

  svg {
    margin-right: 5px;
  }
`;

interface IProps {
  thumbnail: string;
  name: string;
  description: string;
  city: string;
}

const PlaceCard: React.FC<IProps> = ({ thumbnail, name, description, city }) => (
  <Card>
    <ImageContainer>
      <img src={thumbnail} alt="thumbnail" />
      <span>Lihat Detail</span>
    </ImageContainer>
    <Content>
      <h3>{name}</h3>
      <p>{description}</p>
      <p className="city"><MdiCityIcon /> {city}</p>
    </Content>
  </Card>
);

export default PlaceCard;
