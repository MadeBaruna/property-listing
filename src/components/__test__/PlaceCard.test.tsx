import React from 'react';
import 'jest-styled-components'
import { render } from 'enzyme';
import { MemoryRouter } from 'react-router';

import PlaceCard from '../PlaceCard';

it('renders <PlaceCard />', () => {
  const wrapper = render(
    <MemoryRouter>
      <PlaceCard id="randomid" thumbnail="https://loremflickr.com/320/240/apartment,office?lock=1" name="Mediterania Garden Residence 22222222" description="Mediterania Garden Residence 2 berlokasi di Podomoro City Jl. Letjen S.Parman Kav.28. Apartemen ini dikembangkan oleh PT Agung Podomoro Land , Tbk,Mediterania Garden Residence 2 memiliki 6 tower. Dalam kompleks ini tersedia beragam fasilitas untuk mendukung hunian Anda, Terdapat pula beragam sarana dan fasilitas yang menjamin kenyamanan hunian Anda, seperti: Access Card, Atm Center, Basketball Court, Bbq Pit, Drug Store, Function Hall, Gym, Jogging Track, Laundry, Mini Market, Movie Theater, Playground, Restaurant, Salon, Super Market, Swimming Pool dan Tennis Court." city="Jakarta Barat" />
    </MemoryRouter>
  );

  expect(wrapper).toMatchSnapshot();
});
