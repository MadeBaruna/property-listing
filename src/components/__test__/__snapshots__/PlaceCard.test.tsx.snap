// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <PlaceCard /> 1`] = `
.c2 {
  border: none;
  border-radius: 4px;
  padding: 6px 12px;
  line-height: 24px;
  font-size: 16px;
  background: #04ACF4;
  color: white;
  cursor: pointer;
  font-family: 'Segoe UI',Tahoma,Geneva,Verdana,sans-serif;
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c2.outlined {
  background: white;
  border: 1px solid #04ACF4;
  color: #04ACF4;
}

.c2.outlined:hover {
  background: #04ACF4;
  color: white;
}

.c2:hover {
  background: #0484f4;
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: white;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: calc(50% - 20px);
  max-width: 320px;
  margin: 10px;
  cursor: pointer;
  -webkit-transition: 100ms ease-out;
  transition: 100ms ease-out;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

.c0 img {
  object-fit: cover;
  width: 100%;
}

.c0:hover {
  box-shadow: 0px 13px 21px -5px rgba(0,0,0,0.2);
  -webkit-transition: 100ms ease-out;
  transition: 100ms ease-out;
}

.c1 {
  padding: 10px;
}

.c1 h3 {
  white-space: nowrap;
  overflow: hidden;
  display: block;
  text-overflow: ellipsis;
}

.c1 p {
  height: 100%;
  line-height: 18px;
  max-height: 36px;
  word-wrap: break-word;
  overflow: hidden;
  margin-top: 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 p.city {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 5px;
}

.c1 svg {
  margin-right: 5px;
}

<div
  class="c0"
>
  <img
    alt="thumbnail"
    src="https://loremflickr.com/320/240/apartment,office?lock=1"
  />
  <div
    class="c1"
  >
    <h3>
      Mediterania Garden Residence 22222222
    </h3>
    <p>
      Mediterania Garden Residence 2 berlokasi di Podomoro City Jl. Letjen S.Parman Kav.28. Apartemen ini dikembangkan oleh PT Agung Podomoro Land , Tbk,Mediterania Garden Residence 2 memiliki 6 tower. Dalam kompleks ini tersedia beragam fasilitas untuk mendukung hunian Anda, Terdapat pula beragam sarana dan fasilitas yang menjamin kenyamanan hunian Anda, seperti: Access Card, Atm Center, Basketball Court, Bbq Pit, Drug Store, Function Hall, Gym, Jogging Track, Laundry, Mini Market, Movie Theater, Playground, Restaurant, Salon, Super Market, Swimming Pool dan Tennis Court.
    </p>
    <p
      class="city"
    >
      <svg
        class="mdi-icon "
        fill="currentColor"
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M19,15H17V13H19M19,19H17V17H19M13,7H11V5H13M13,11H11V9H13M13,15H11V13H13M13,19H11V17H13M7,11H5V9H7M7,15H5V13H7M7,19H5V17H7M15,11V5L12,2L9,5V7H3V21H21V11H15Z"
        />
      </svg>
       Jakarta Barat
    </p>
    <button
      class="outlined c2"
      style="float:right"
    >
      Lihat Detail
    </button>
  </div>
</div>
`;
